### PR TITLE
feat: new properties for send-mail delivery provider

### PR DESCRIPTION
**Pull request by macrozone**

### CODE REVIEW

File: packages/core-delivery/plugins/send-mail.js [hunk 1]
```diff
@@ -25,11 +25,13 @@ class SendMail extends DeliveryAdapter {
     return type === 'SHIPPING';
   }
 
-  isActive() { // eslint-disable-line
+  isActive() {
+    // eslint-disable-line
```
Comment (pozylon) [inline packages/core-delivery/plugins/send-mail.js]: This will lead to more warnings I guess :)

Comment (macrozone) [reply packages/core-delivery/plugins/send-mail.js]: not sure why the reformat ;-)

Comment (pozylon) [reply packages/core-delivery/plugins/send-mail.js]: prettier but shittier 🤟

**Comment by pozylon** [comment]
What if we just forward the whole order object to the template? It would be more generic and totally delegate what data to show and how to format to the actual template.

**Comment by pozylon** [comment]
```
return director.sendMessage({
      template: 'shop.unchained.send-mail',
      attachments,
      meta: {
        mailPrefix: `${order.orderNumber}_`,
        from: this.getFromAddress(),
        to: this.getToAddress(),
        cc: this.getCCAddress(),
        transactionContext,
        order
      }
    });
```

**Comment by macrozone** [comment]
we should also rename meta in all emails, because meta is ambiguos.

i would just call it `props`

forwarding order does not work so good, because the template language used can't resolve the helpers.  I would also suggest to swtich the template language as mustache is too limited in my opinion. Better use handlebars or so. Maybe we can even use blaze or handlebars in a way, so that object helpers are resolved, which would be very nice

**Comment by pozylon** [comment]
Ok so maybe more like this without props and meta:

```
return director.sendMessage({
      template: 'shop.unchained.send-mail',
      attachments,
      mailPrefix: `${order.orderNumber}_`,
      from: this.getFromAddress(),
      to: this.getToAddress(),
      cc: this.getCCAddress(),
    }, { transactionContext,
        order
});
```

Yeah, if handlebars can do that, it would match much better.

**Comment by pozylon** [comment]
Going to merge this, let's do the "Refactor E-Mail" in another PR

**Pull request merged**